frontend/src/metabase/entities/model-indexes/actions.ts

```diff
@@ -1,34 +1,46 @@
 import { dissocIn } from "icepick";
 import _ from "underscore";
 
+import {
+  createModelIndex,
+  deleteModelIndex,
+  listModelIndexes,
+} from "metabase/api";
 import type Question from "metabase-lib/v1/Question";
 import type { FieldReference, Field } from "metabase-types/api";
 import type { ModelIndex } from "metabase-types/api/modelIndexes";
 import type { Dispatch } from "metabase-types/store";
 
-import { ModelIndexes } from "./model-indexes";
+import { getPkRef } from "./utils";
 
 export type FieldWithMaybeIndex = Field & {
   should_index?: boolean;
-  field_ref?: FieldReference;
+};
+
+type FieldWithIndexAndRef = Field & {
+  should_index: boolean;
+  field_ref: FieldReference;
+};
+
+const filterForRefAndShouldIndex = (
+  field: FieldWithMaybeIndex,
+): field is FieldWithIndexAndRef => {
+  return field.should_index !== undefined && !!field.field_ref;
 };
 
 export const updateModelIndexes =
-  (model: Question) => async (dispatch: Dispatch, getState: any) => {
+  (model: Question) => async (dispatch: Dispatch) => {
     const fields = model.getResultMetadata();
 
-    const fieldsWithIndexFlags = fields.filter(
-      (field: FieldWithMaybeIndex) => field.should_index !== undefined,
-    );
+    const fieldsWithIndexFlags = fields.filter(filterForRefAndShouldIndex);
 
     if (fieldsWithIndexFlags.length === 0) {
       return;
     }
 
-    const existingIndexes: ModelIndex[] =
-      ModelIndexes.selectors.getList(getState(), {
-        entityQuery: { model_id: model.id() },
-      }) ?? [];
+    const { data: existingIndexes = [] } = await (dispatch(
+      listModelIndexes.initiate({ model_id: model.id() }),
+    ) as Promise<{ data: ModelIndex[] }>);
```

Comment:
would it be better to throw if there's a failure? i think we're likely to generate to failed/incorrect requests below if this requests fails.